cdap-common/src/main/java/co/cask/cdap/common/kerberos/DefaultOwnerAdmin.java

```diff
@@ -80,10 +76,35 @@ public ImpersonationInfo getImpersonationInfo(NamespacedEntityId entityId) throw
       }
     }
     // (CDAP-8176) Since no owner was found for the entity return namespace principal if present.
+    NamespaceConfig nsConfig = getNamespacePrincipal(entityId.getNamespaceId());
```

Comment:
Is that method ill-named? It is called getPrincipal but returns a namespace config.